x-pack/plugins/security_solution/public/detection_engine/rule_management/hooks/use_alert_suppression.tsx

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import type { Type } from '@kbn/securitysolution-io-ts-alerting-types';
+import { isSuppressibleAlertRule } from '../../../../common/detection_engine/utils';
+import { useIsExperimentalFeatureEnabled } from '../../../common/hooks/use_experimental_features';
+export interface UseAlertSuppressionReturn {
+  isSuppressionEnabled: boolean;
+}
+
+export const useAlertSuppression = (ruleType?: Type): UseAlertSuppressionReturn => {
```

Comment:
Why the argument is optional?

Reply:
some of the consumers of the hook, might have rule type `undefined`. 
I suspect this might be a reason. cc: @WafaaNasr 

I redefined it as `Type | undefined` instead